subsys/bluetooth/rpc/client/bt_rpc_internal_client.c

```diff
@@ -32,3 +36,73 @@ bool bt_addr_le_is_bonded(uint8_t id, const bt_addr_le_t *addr)
 
 	return result;
 }
+
+struct bt_hci_cmd_send_sync_res {
+	int result;
+	struct net_buf *buf;
+	struct net_buf **rsp;
+};
+
+static void decode_net_buf(struct ser_scratchpad *scratchpad, struct net_buf *data)
+{
+	size_t len;
+	void *buf;
+
+	buf = ser_decode_buffer_into_scratchpad(scratchpad, &len);
+	net_buf_add_mem(data, buf, len);
+}
+
+static void bt_hci_cmd_send_sync_rsp(const struct nrf_rpc_group *group,
+				     struct nrf_rpc_cbor_ctx *ctx, void *handler_data)
+{
+	struct bt_hci_cmd_send_sync_res *res;
+	struct ser_scratchpad scratchpad;
+
+	res = (struct bt_hci_cmd_send_sync_res *)handler_data;
+
+	res->result = ser_decode_int(ctx);
+
+	if (ser_decode_is_null(ctx)) {
+		*res->rsp = NULL;
+	} else {
+		SER_SCRATCHPAD_DECLARE(&scratchpad, ctx);
+		*res->rsp = net_buf_alloc(&hci_cmd_pool, K_FOREVER);
```

Comment:
Why we need command pool here? Where is this buffer freed?

Reply:
Function `bt_hci_cmd_send_sync` can be called with non-NULL `rsp`, if the caller is interested in response parameters.
The caller is responsible for calling `net_buf_unref()` on the buffer when done parsing it.